sklearn/neighbors/kde.py

```diff
@@ -124,6 +130,11 @@ def fit(self, X, y=None, sample_weight=None):
         sample_weight : array_like, shape (n_samples,), optional
             List of sample weights attached to the data X.
         """
+
+        if self.metric == 'minkowski' and self.p != 2:
```

Comment:
you should only change the given `p` if it's `warn`, and leave as is if it's explicitly set by the user.